Dockerfile

```
@@ -3,7 +3,7 @@ WORKDIR /app
 COPY ./package.json ./
 COPY ./dist ./dist
 COPY ./.yarn ./.yarn
-COPY ./.pnp.js ./
+COPY ./.pnp.cjs ./
 COPY ./yarn.lock ./
 EXPOSE 3000
 CMD [ "yarn", "start:prod" ]
```

```suggestion
CMD [ "node", "-r", "./.pnp.cjs", "dist/main" ]
```

It's more efficient to start node directly

---

More documentation on this: https://yarnpkg.com/features/pnp#initializing-pnp (We should probably link that from here, if we apply this change.)

@longtomjr, what do you think?

---

Sounds good. If this reduce the overhead we should use node and link to the docs.

---

I wouldn't think the overhead is a big concern in terms of performance (starting up the server is unlikely to become a bottleneck 😄), but avoiding the need to indirect through yarn is potentially a significant reduction in complexity for the deployable image, if we can omit yarn entirely, in favour of just including its output? We should maybe have a separate PR dedicated to that.

---

Not sure if the cost is only for server startup, but I am happy either way. Does not seem like a massive change for this PR, and then we can have a backlog task / follow up pr to look at the deployed image complexity.

---

It should only for startup: according to the docs, all `yarn` does is to add that flag before invoking `node`.